src/node-binance-api.ts

```diff
@@ -638,6 +645,42 @@ export default class Binance {
         }
     }
 
+    unarmorKey(a:string):number[] {
+        // eslint-disable-next-line no-useless-escape
+        const m = /-----BEGIN [^-]+-----\n([A-Za-z0-9+\/=\s]+)\n-----END [^-]+-----|begin-base64[^\n]+\n([A-Za-z0-9+\/=\s]+)====/.exec(a);
+        if (m) {
+            if (m[1]) {
+                a = m[1];
+            } else if (m[2]) {
+                a = m[2];
+            } else {
+                throw new Error("RegExp out of sync");
+            }
+        }
+        return base64.decode(a);
+    }
+
+    generateSignature(query: string) {
+        const APISECRET = this.Options.APISECRET || this.APISECRET;
+        let signature = '';
+        if (APISECRET.includes ('PRIVATE KEY')) {
+            // if less than the below length, then it can't be RSA key
+            if (APISECRET.length < 500) {
+                const privateKey = new Uint8Array (this.unarmorKey (APISECRET).slice (16));
+                const encodedQuery = new TextEncoder().encode(query);
+                const signatureInit = ed25519.sign (encodedQuery, privateKey);
+                signature = base64.encode (signatureInit);
+            } else {
+                const signed = createSign('RSA-SHA256').update(query);
+                signature = signed.sign(APISECRET, 'base64');
+                signature = encodeURIComponent (signature);
+            }
+        } else {
+            signature = crypto.createHmac('sha256', this.Options.APISECRET).update(query).digest('hex'); // set the HMAC hash header
```

Comment:
we shouldn't use `this.Options`, but directly `this.APISECRET` 

besides that, we should also support `this.PRIVATEKEY`